aicspylibczi/CziFile.py

```diff
@@ -261,15 +266,25 @@ def read_subblock_metadata(self, **kwargs):
 
         Returns
         -------
-        [str]
-            an array of stings containing the specified subblock metadata
-
+        [(dict, str)] if unified_xml is False
+            an array of tuples containing a dimension dictionary and the corresponding subblock metadata
+        lxml.etree.Element if unified_xml is True
+            an lxml document containing the requested metadata
         """
         plane_constraints = self.czilib.DimCoord()
         [plane_constraints.set_dim(k, v) for (k, v) in kwargs.items() if k in CziFile.ZISRAW_DIMS]
         m_index = self._get_m_index_from_kwargs(kwargs)
-
-        return self.reader.read_meta_from_subblock(plane_constraints, m_index)
+        subblock_meta = self.reader.read_meta_from_subblock(plane_constraints, m_index)
+        if not unified_xml:
+            return subblock_meta
+        root = etree.Element("Subblocks")
+        for pair in subblock_meta:
+            new_element = etree.Element("Subblock")
+            for dim, number in pair[0].items():
+                new_element.set(dim, str(number))
+            new_element.append(etree.XML(pair[1]))
+            root.append(new_element)
+        return root
```

Comment:
so the challenge I have is the original function returns something like 
```
[({'S'=0, 'C'=1, ...}, <METADATA><Tags><AcquisitionTime>2019-06-27T18:33:41.1154211Z</AcquisitionTime>), .....]
```
The above structure strikes me as overly complicated now that I've merged the dims as sunblock attributes. 

The unified_xml=True result would either be an lxml.etree.Element or a string. Neither matches the signature of the first. 
My thought is that when the package hits 3.0 to drop the original return type plus now that `def meta` includes the subblock metadata I don't forsee people actually using the `def read_subblock_metadata`.

Thoughts on this?

Reply:
I decided to force the S dimensions on the Subblock. It's purely a convenience for XSLT. If it's not there I have to add a lot of messy logic to the transform sheets.

Reply:
> Thoughts on this?

I could be confused about this, but I thought we had to send xml metadata up as strings and not as xml trees because of dask distributed serialization.

Reply:
> I decided to force the S dimensions on the Subblock. It's purely a convenience for XSLT. If it's not there I have to add a lot of messy logic to the transform sheets.

The implication of this is that your XSLT will not be converting pure Zeiss CZI xml -- it will only be able to convert xml read by aicspylibczi.

Reply:
^ This is a really interesting point

Reply:
You make a good point @toloudis. I did it because I thought, in general, it would be advantageous for the user and the subblock metadata is key information especially when we convert to OME. After thinking about it I will return the subblock metadata in a separate string in have the XSLT merge them. Unfortunately, the argument could still be made that the user would have to use AICSImage because the context is not captured in the subblock metadata, it's implicit. To work around this I had to figure out a way to serialize this but it's not part of the standard. I will put in a mention on ImageSC to Zeiss recommending that libCZI provide a function to export all the subblock metadata and suggest a format then we would have a standard that might cross different processing languages. I'll try and get the changes in this evening.